FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c

```diff
@@ -94,146 +94,223 @@ extern void vTaskEnterCritical(void);
-    #define configUSE_TASK_PREEMPTION_DISABLE               1
-
-Coverage for 
-        static void vTaskPreemptionEnable( void );
-        covers the deafult state when xSchedulerRunning is set to False
-*/
-void test_task_preemption_enable_branch_xSchedulerRunning_False( void )
+/**
+ * @brief vTaskPreemptionEnable - Enable preemption of a task when scheduler is running.
+ *
+ * The xPreemptionDisable of the task will be set to pdFALSE.
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ * if( xSchedulerRunning != pdFALSE )
+ * {
+ *     if( taskTASK_IS_RUNNING( pxTCB ) == pdTRUE )
+ *     {
+ *         xCoreID = ( BaseType_t ) pxTCB->xTaskRunState;
+ *         prvYieldCore( xCoreID );
+ *     }
+ * }
+ * @endcode
+ * ( xSchedulerRunning != pdFALSE ) is true.
+ * ( taskTASK_IS_RUNNING( pxTCB ) == pdTRUE ) is false.
+ */
+void test_coverage_vTaskPreemptionEnable_scheduler_running( void )
 {
-    TaskHandle_t xTaskHandles[configNUMBER_OF_CORES] = { NULL };
-    uint32_t i;
-
-    /* Create tasks of equal priority */
-    for (i = 0; i < configNUMBER_OF_CORES; i++) {
-        xTaskCreate( vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 1, &xTaskHandles[i] );
-    }
-
-    //Tasks are created and a task is passed but scheduler is never ran
-    vTaskPreemptionEnable( xTaskHandles[0] );
+    TCB_t xTaskTCB = { NULL };
 
-} 
+    /* Setup variable. */
+    xTaskTCB.xPreemptionDisable = pdTRUE;
+    xTaskTCB.xTaskRunState = -1; /* taskTASK_NOT_RUNNING. */
 
-/*
-The kernel will be configured as follows:
-    #define configNUMBER_OF_CORES                               (N > 1)
-    #define configUSE_CORE_AFFINITY                         1
-    #define configUSE_TASK_PREEMPTION_DISABLE               1
+    xSchedulerRunning = pdTRUE;
+    
+    /* Clear callback in commonSetUp. */
+    vFakePortEnterCriticalSection_StubWithCallback( NULL );
+    vFakePortExitCriticalSection_StubWithCallback( NULL );
 
-Coverage for 
-        static void vTaskPreemptionEnable( void );
-        covers the deafult state when NULL task is passed
-*/
-void test_task_preemption_enable_branch_NULL_task( void )
-{
-    TaskHandle_t xTaskHandles[configNUMBER_OF_CORES] = { NULL };
-    uint32_t i;
+    /* Expections. */
+    vFakePortEnterCriticalSection_Expect();
+    vFakePortExitCriticalSection_Expect();
 
-    /* Create tasks of equal priority */
-    for (i = 0; i < configNUMBER_OF_CORES; i++) {
-        xTaskCreate( vSmpTestTask, "SMP Task", configMINIMAL_STACK_SIZE, NULL, 1, &xTaskHandles[i] );
-    }
+    /* API call. */
+    vTaskPreemptionEnable( &xTaskTCB );
 
-    vTaskStartScheduler();
+    /* Validation. */
+    TEST_ASSERT( xTaskTCB.xPreemptionDisable == pdFALSE );
+}
 
-    /* task T0 */
+/**
+ * @brief vTaskPreemptionEnable - Enable preemption of a task with NULL handle.
+ *
+ * The xPreemptionDisable of the task on core 0 will be set to pdFALSE.
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ * pxTCB = prvGetTCBFromHandle( xTask );
```

Comment:
Off topic for this review, but it dosnt look like this function checks the return from prvGetTCBFromHandle() to verify pxTCB is not NULL before use, it probably should ...

Reply:
The implementation of this function is
```
#define prvGetTCBFromHandle( pxHandle )    ( ( ( pxHandle ) == NULL ) ? pxCurrentTCB : ( pxHandle ) )
```
The implementation ensures that pxCurrentTCB can't be NULL after scheduler is running.